components/settings/projects/ProjectsSettings.tsx

```
@@ -33,49 +52,121 @@ function ProjectRow({
-    mutate(
-      (projects) => {
-        if (!projects || !projectWithMembers) {
-          return projects;
-        }
+    if (isTeamLead) {
+      const updatedProject = {
+        ...projectValues,
+        id: projectWithMembers.id,
+        projectMembers: projectValues.projectMembers.map((member, index) => ({
+          ...projectWithMembers.projectMembers[index],
+          ...member
+        }))
+      };
+
+      updateProject(updatedProject);
 
-        return projects.map((_project) => {
-          if (_project.id === projectWithMembers.id) {
-            return {
-              ..._project,
-              ...updatedProject,
-              projectMembers: _project.projectMembers.map((projectMember, index) => {
-                return {
-                  ...projectMember,
-                  ...updatedProject.projectMembers[index]
-                };
-              })
-            };
+      form.reset(updatedProject, {
+        keepDirty: false
+      });
+
+      mutate(
+        (projects) => {
+          if (!projects || !projectWithMembers) {
+            return projects;
           }
 
-          return _project;
-        });
-      },
-      {
-        revalidate: false
+          return projects.map((_project) => {
+            if (_project.id === projectWithMembers.id) {
+              return {
+                ..._project,
+                ...updatedProject,
+                projectMembers: _project.projectMembers.map((projectMember, index) => {
+                  return {
+                    ...projectMember,
+                    ...updatedProject.projectMembers[index]
+                  };
+                })
+              };
+            }
+
+            return _project;
+          });
+        },
+        {
+          revalidate: false
+        }
+      );
+    } else {
+      const projectMemberValue = projectValues.projectMembers.find((member) => member.userId === user?.id);
+      if (!projectMemberValue || !projectMemberValue.id) {
```

is this for authentication? I think i would expect that we just call the backend and show the user an error if it fails. Right now this just silently fails

---

Its not just authentication, this will ideally never happen. Its mostly to make typescript happy, otherwise I would need to use !. Its probably looking weird due to the return statement.